Webhooks/SimpleTexting.ashx

```diff
@@ -0,0 +1,524 @@
+
+        switch ( unsubscribeReport.Type )
+        {
+            case SimpleTextingDotNet.Enum.Trigger.INCOMING_MESSAGE:
+            case SimpleTextingDotNet.Enum.Trigger.OUTGOING_MESSAGE:
+                messageReport = JsonConvert.DeserializeObject<MessageReport>( payload );
+                unsubscribeReport = null;
+                break;
+            case SimpleTextingDotNet.Enum.Trigger.DELIVERY_REPORT:
+            case SimpleTextingDotNet.Enum.Trigger.NON_DELIVERED_REPORT:
+                deliveryReport = JsonConvert.DeserializeObject<DeliveryMessageReport>( payload );
+                unsubscribeReport = null;
+                break;
+        }
+
+        if ( unsubscribeReport == null && messageReport == null && deliveryReport == null )
+        {
+            response.Write( "Invalid content type." );
+            response.StatusCode = ( int ) System.Net.HttpStatusCode.NotAcceptable;
+            return retObj;
+        }
+
+        if ( unsubscribeReport != null )
+        {
+            retObj = unsubscribeReport;
+        }
+        else if ( messageReport != null )
+        {
+            retObj = messageReport;
+        }
+        else if ( deliveryReport != null )
+        {
+            retObj = deliveryReport;
+        }
+
+        response.StatusCode = ( int ) System.Net.HttpStatusCode.OK;
+        return retObj;
+    }
+
+
+    /// <summary>
+    /// Mark message as not delivered.
+    /// </summary>
+    private void MessageNotDelivered( DeliveryMessageReport deliveryReport )
+    {
+        if ( deliveryReport != null && deliveryReport.Values != null && deliveryReport.Values.MessageId.IsNotNullOrWhiteSpace() )
+        {
+            var messageId = deliveryReport.Values.MessageId;
+
+            using ( RockContext rockContext = new RockContext() )
+            {
+                CommunicationRecipientService recipientService = new CommunicationRecipientService( rockContext );
+
+                var communicationRecipient = recipientService
+                    .Queryable()
+                    .Where( r => r.UniqueMessageId == messageId )
+                    .FirstOrDefault();
+
+                if ( communicationRecipient != null )
+                {
+                    communicationRecipient.Status = CommunicationRecipientStatus.Failed;
+                    communicationRecipient.StatusNote = $"Message failure notification from Simple Texting on {RockDateTime.Now.ToString()}. Carrier: {deliveryReport.Values.Carrier}";
+                    rockContext.SaveChanges();
+                }
+                else
+                {
+                    ExceptionLogService.LogException( $"No recipient was found with the specified MessageId value! ({messageId})" );
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Mark message as delivered.
+    /// </summary>
+    private void MessageDelivered( DeliveryMessageReport deliveryReport )
```

Comment:
Looks like a lot of shared code with the method above. Can they be simplified to a single method?